actionmailer/test/base_test.rb

```diff
@@ -618,46 +618,61 @@ def self.delivered_email(mail)
     end
   end
 
-  test "you can register an observer to the mail object that gets informed on email delivery" do
-    mail_side_effects do
-      ActionMailer::Base.register_observer(MyObserver)
-      mail = BaseMailer.welcome
-      assert_called_with(MyObserver, :delivered_email, [mail]) do
-        mail.deliver_now
-      end
+  test "you can register and unregister an observer to the mail object that gets informed on email delivery" do
+    ActionMailer::Base.register_observer(MyObserver)
+    mail = BaseMailer.welcome
+    assert_called_with(MyObserver, :delivered_email, [mail]) do
+      mail.deliver_now
+    end
+
+    ActionMailer::Base.unregister_observer(MyObserver)
```

Comment:
This needs to be called inside the `ensure` otherwise if the test fail in the lines above this the observer will still be registered.

Reply:
Thanks for your review. I'll fix it as soon as possible.

Reply:
@rafaelfranca 
How about [this](1286c12).